openslides/core/models.py

```diff
@@ -1,62 +1,95 @@
         """
-        General permissions that can not be placed at a specific app.
+        Contains general permissions that can not be placed in a specific app.
         """
         permissions = (
-            ('can_manage_projector', ugettext_noop('Can manage the projector')),
             ('can_see_projector', ugettext_noop('Can see the projector')),
+            ('can_manage_projector', ugettext_noop('Can manage the projector')),
             ('can_see_dashboard', ugettext_noop('Can see the dashboard')),
-            ('can_use_chat', ugettext_noop('Can use the chat')),
-        )
+            ('can_use_chat', ugettext_noop('Can use the chat')))
+
+    @property
+    def projector_elements(self):
+        """
+        A generator to retrieve all projector elements given in the config
+        field. For every element the method get_api_data() is called and its
+        result returned.
+        """
+        elements = {}
+        for element in ProjectorElement.get_all():
+            elements[element.name] = element
+        for entry in self.config:
+            element = elements.get(entry['name'])
+            if element is None:
+                data = {'name': entry['name'], 'error': 'Projector element does not exist.'}
+            else:
+                try:
+                    data = element.get_api_data(
+                        projector_object=self,
+                        config_entry=entry)
+                except ProjectorException as e:
+                    data = {'name': entry['name'], 'error': str(e)}
+            yield data
+
+
+class CustomSlide(RESTModelMixin, AbsoluteUrlMixin, models.Model):
+    """
+    Model for slides with custom content.
+    """
+    title = models.CharField(
+        verbose_name=ugettext_lazy('Title'),
+        max_length=256)
+    text = models.TextField(
+        verbose_name=ugettext_lazy('Text'),
+        blank=True)
+    weight = models.IntegerField(
+        verbose_name=ugettext_lazy('Weight'),
+        default=0)
+
+    class Meta:
+        ordering = ('weight', 'title', )
 
     def __str__(self):
         return self.title
 
-    def get_absolute_url(self, link='update'):
-        if link == 'update':
-            url = reverse('customslide_update', args=[str(self.pk)])
-        elif link == 'delete':
-            url = reverse('customslide_delete', args=[str(self.pk)])
-        else:
-            url = super().get_absolute_url(link)
-        return url
-
 
 class Tag(RESTModelMixin, AbsoluteUrlMixin, models.Model):
     """
-    Model to save tags.
+    Model for tags. This tags can be used for other models like agenda items,
+    motions or assignments.
     """
-
-    name = models.CharField(max_length=255, unique=True,
-                            verbose_name=ugettext_lazy('Tag'))
+    name = models.CharField(
+        verbose_name=ugettext_lazy('Tag'),
+        max_length=255,
+        unique=True)
 
     class Meta:
-        ordering = ['name']
+        ordering = ('name',)
```

Comment:
I am not sure. Django still uses tuples in Meta.ordering in there contrib apps. This was only a change for the config. Maybe they will make a decision for "meta-config" to in the next future. I would say we wait until then.